### PR TITLE
Updated code and templates to be Python 3.x compatible.

### PR DESCRIPTION
**Pull request by ddimensia**
Pulled in some Python 3.x changes from the originating repo, added fixes to templates and a few other locations that were still not Python 3 compatible.

### CODE REVIEW

File: src/emitters/templates/py_interface.jinja2 [hunk 1]
```diff
@@ -232,18 +232,10 @@ class XBVCEdgePoint(object):
 
     def stop(self):
         self._stop_event.set()
-        try:
-            self._recv_thread_handle.join()
-        except:
-            pass
-        
+
+        self._recv_thread_handle.join()
```
Comment (brentpicasso) [inline src/emitters/templates/py_interface.jinja2]: why remove the exception catching?

File: src/emitters/templates/py_interface.jinja2 [hunk 2]
```diff
@@ -232,18 +232,10 @@ class XBVCEdgePoint(object):
...
         self._recv_thread_handle = None
-        
-        try:
-            self._dispatch_thread_handle.join()
-        except:
-            pass
-        
+        self._dispatch_thread_handle.join()
```
Comment (brentpicasso) [inline src/emitters/templates/py_interface.jinja2]: ditto

**Comment by brentpicasso** [comment]
Looks like pretty straightforward syntax change, just one question.

File: src/emitters/templates/py_interface.jinja2 (hunk 1 above)
Comment (ddimensia) [reply src/emitters/templates/py_interface.jinja2]: Actually not 100% sure, this is a change that I pulled down from original repo.  I can add it back in tonight.

Comment (brentpicasso) [reply src/emitters/templates/py_interface.jinja2]: Yes, let's keep the logic exactly as we have it in our forked repo - the RC app and upstream code may depend on this behavior. Thanks!

Comment (ddimensia) [reply src/emitters/templates/py_interface.jinja2]: Fixed

File: src/emitters/templates/py_interface.jinja2 (hunk 2 above)
Comment (ddimensia) [reply src/emitters/templates/py_interface.jinja2]: Fixed

**Pull request merged**